api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -200,111 +275,103 @@ def publish_cdr(cmd_name, args)
-  account = env.fetch(:publisher_account)
-  app_sa = "#{op.opts.project}@appspot.gserviceaccount.com"
   tier = env.fetch(:accessTiers)[op.opts.tier]
-
-  # TODO(RW-3208): Investigate using a temporary / impersonated SA credential instead of a key.
-  key_file = Tempfile.new(["#{account}-key", ".json"], "/tmp")
-  ServiceAccountContext.new(
-    op.opts.project, account, key_file.path).run do
-    # TODO(RW-3768): This currently leaves the user session with an activated service
-    # account user. Ideally the activation would be hermetic within the docker
-    # session, or else we would revert the active account after running.
-    common.run_inline %W{gcloud auth activate-service-account -q --key-file #{key_file.path}}
-
-    source_cdr_project = op.opts.source_cdr_project_override || tier.fetch(:source_cdr_project)
-    source_dataset = "#{source_cdr_project}:#{op.opts.bq_dataset}"
-    ingest_dataset = "#{tier.fetch(:ingest_cdr_project)}:#{op.opts.bq_dataset}"
-    dest_dataset = "#{tier.fetch(:dest_cdr_project)}:#{op.opts.bq_dataset}"
-    common.status "Copying from '#{source_dataset}' -> '#{ingest_dataset}' -> '#{dest_dataset}' as #{account}"
-
-    # If you receive an error from "bq" like "Invalid JWT Signature", you may
-    # need to delete cached BigQuery creds on your local machine. Try running
-    # bq init --delete_credentials as recommended in the output.
-    # TODO(RW-3768): Find a better solution for Google credentials in docker.
-
-    # Copy through an intermediate project and delete after (include TTL in case later steps fail).
-    # See https://docs.google.com/document/d/1EHw5nisXspJjA9yeZput3W4-vSIcuLBU5dPizTnk1i0/edit
-    common.run_inline %W{bq mk -f --default_table_expiration 7200 --dataset #{ingest_dataset}}
-    common.run_inline %W{./copy-bq-dataset.sh
-        #{source_dataset} #{ingest_dataset} #{source_cdr_project}
-        #{table_match_filter} #{table_skip_filter}}
-
-    common.run_inline %W{bq mk -f --dataset #{dest_dataset}}
-    common.run_inline %W{./copy-bq-dataset.sh
-        #{ingest_dataset} #{dest_dataset} #{tier.fetch(:ingest_cdr_project)}
-        #{table_match_filter} #{table_skip_filter}}
-
-    # Delete the intermediate dataset.
-    common.run_inline %W{bq rm -r -f --dataset #{ingest_dataset}}
-
-    auth_domain_group_email = tier.fetch(:auth_domain_group_email)
-
-    config_file = Tempfile.new("#{op.opts.bq_dataset}-config.json")
-    begin
-      json = JSON.parse(
-        common.capture_stdout %{bq show --format=prettyjson #{dest_dataset}})
-      existing_groups = Set[]
-      existing_users = Set[]
-      for entry in json["access"]
-        if entry.key?("groupByEmail")
-          existing_groups.add(entry["groupByEmail"])
-        end
-        if entry.key?("userByEmail")
-          existing_users.add(entry["userByEmail"])
-        end
+  source_cdr_project = tier.fetch(:source_cdr_project)
+  dest_dataset = "#{tier.fetch(:dest_cdr_project)}:#{op.opts.bq_dataset}"
```

Comment:
nit: broad spec ops uses a pattern of prefixing `fq_` for datasets/tables that are fully qualified which I thought was pretty useful

Reply:
Switched to this pattern